manifests/install.pp

```diff
@@ -15,37 +15,50 @@
 
   if $consul::install_method == 'url' {
 
+    $version = $consul::version
+    $consul_archive_target = "consul_${version}.zip"
+    $consul_web_ui_archive_target = "consul_web_ui_${version}.zip"
+
     if $::operatingsystem != 'darwin' {
       ensure_packages(['unzip'])
     }
-    staging::file { 'consul.zip':
+    staging::file { $consul_archive_target:
       source => $consul::download_url
     } ->
-    staging::extract { 'consul.zip':
-      target  => $consul::bin_dir,
-      creates => "${consul::bin_dir}/consul",
+    file { "${consul::bin_dir}/consul_${version}/":
+      ensure => 'directory',
+      owner  => 'root',
+      group  => 0,
+      mode   => '0755',
+    } ->
+    staging::extract { $consul_archive_target:
+      target  => "${consul::bin_dir}/consul_${version}/",
+      creates => "${consul::bin_dir}/consul_${version}/consul",
     } ->
     file { "${consul::bin_dir}/consul":
-      owner => 'root',
-      group => 0, # 0 instead of root because OS X uses "wheel".
-      mode  => '0555',
+      owner  => 'root',
+      group  => 0, # 0 instead of root because OS X uses "wheel".
+      mode   => '0555',
+      target => "${consul::bin_dir}/consul_${version}/consul",
+      notify => Service['consul'],
```

Comment:
This will break puppet if `$consul::manage_service` is `false`. I would suggest notifying the class, like so: `Class['consul::run_service']`